Francesco transport runtime

Gamaranto opened the pull request:
This PR adds the Transport needed to fetch data from the runtime storage. It also creates React Context to be used in components that needs data from the runtime.

## What to Expect
The Transport should be fully implemented and all methods functional. Methods can be tested in the JavaScript console on the local version of Pioneer. 

## What to Not Expect

The is a cache that was implemented initially but never used, it is safe to ignore it. Also the Context is currently not used anywhere, but there is a separate PR for that (To be opened soon)

Lezek123 commented on packages/joy-proposals/src/stories/ProposalDetails.stories.tsx:
```diff
@@ -1,25 +1,25 @@
 };
 
-export const HasToVote = () => <ProposalDetails {...MockProposalDetails} />;
+export const HasToVote = () => <ProposalDetails {...mockedDetails} />;
```
Causes: `Cannot find name 'mockedDetails'`

Lezek123 commented on packages/joy-proposals/src/runtime/TransportContext.tsx:
```diff
@@ -0,0 +1,23 @@
+  return <TransportContext.Provider value={new MockTransport()}>{children}</TransportContext.Provider>;
+}
+
+export function SubstrateProvider({ children }: { children: React.ReactChildren }) {
```
It should probably be like in `joy-media/src/TransportContext.tsx`:
`export const SubstrateTransportProvider = (props: React.PropsWithChildren<{}>) => {`
Otherwise it causes some TS errors when trying to wrap components inside it.
Actually, the entire implementation can be just copied from that file.

Lezek123 commented on packages/joy-proposals/src/runtime/transport.substrate.ts:
```diff
@@ -0,0 +1,97 @@
+  }
+
+  get proposalEngine() {
+    return this.api.query.proposalEngine;
```
Should be proposal**s**Engine

Lezek123 commented on packages/joy-proposals/src/runtime/transport.substrate.ts:
```diff
@@ -0,0 +1,97 @@
+  }
+
+  async activeProposals() {
+    const activeProposalsIds = await this.proposalEngine.activeProposalsIds<ProposalId[]>();
```
Should be: `activeProposalIds`

Lezek123 commented on packages/joy-proposals/src/runtime/transport.substrate.ts:
```diff
@@ -0,0 +1,97 @@
+  }
+
+  get proposalCodex() {
+    return this.api.query.codex;
```
Should be: api.query.**proposals**Codex

Gamaranto commented on packages/joy-utils/src/newTxBtn.tsx:
```diff
@@ -0,0 +1,56 @@
+import { SemanticShorthandItem, IconProps, SemanticCOLORS, Button } from "semantic-ui-react";
+
+type InjectedProps = {
+  queueExtrinsic: QueueTxExtrinsicAdd;
```
Removed, had no place in this PR.

Gamaranto commented on packages/joy-proposals/src/stories/ProposalTypes.stories.tsx:
```diff
@@ -0,0 +1,10 @@
+import React from "react";
+import "../index.css";
+import { ProposalTypeSelector } from "../ProposalTypeSelector";
```
Removed, had no place in this PR, was just an old component story.

Gamaranto commented on packages/joy-proposals/src/runtime/transport.substrate.ts:
```diff
@@ -0,0 +1,97 @@
+
+  async proposalTypesGracePeriod() {
+    // Cheating here,we know what the keys are.
+    let methods = excludeKeys(
```
I agree. I changed this to be inclusive rather than exclusive.

Gamaranto commented:
Fixed all typos (or should be all...) in d8de6c4 and refactored `excludeKeys`to `includeKeys` in 075a184 . I've tested it locally and it seems to work here

Lezek123 commented on packages/joy-proposals/src/runtime/TransportContext.tsx:
```diff
@@ -5,19 +5,19 @@ import { SubstrateTransport } from "./transport.substrate";
+const TransportContext = createContext<Transport>((null as unknown) as Transport);
 
-export function MockProvider({ children }: { children: React.ReactChildren }) {
+export function MockProvider({ children }: { children: React.PropsWithChildren<{}> }) {
```
All seems good now, just a little typo here (and a few lines below also):
`({ children }: { children: React.PropsWithChildren<{}> })` should be just: `({ children }: React.PropsWithChildren<{}>`
